subsys/bluetooth/host/conn.c

```diff
@@ -1895,6 +1922,36 @@ int bt_conn_get_info(const struct bt_conn *conn, struct bt_conn_info *info)
 	return -EINVAL;
 }
 
+#if defined(CONFIG_BT_REMOTE_VERSION)
+int hci_le_read_remote_version(struct bt_conn *conn)
+{
+	int err;
+	struct bt_hci_cp_read_remote_version_info *cp;
+	struct net_buf *buf;
+
+	k_sem_init(&conn->rv.sem, 0, 1);
```

Comment:
This doesn't look right. What if there's another `bt_conn_get_info()` call on the same connection that's blocking on `k_sem_take()`? You'd end up messing up the state of the semaphore by calling `k_sem_init()`. The initialisation of the semaphore should happen when the connection object is created, i.e. probably in `conn_new()`.

You're also missing protection for the multiple access like I described above. What might make most sense is to move the `atomic_set_bit(conn->flags, BT_CONN_PENDING_VERSION_INFO);` from below up here, and make it:

```
if (atomic_test_and_set_bit(conn->flags, BT_CONN_PENDING_VERSION_INFO)) {
        return -EBUSY;
}
```